aiida_mlip/calculations/md.py

```diff
@@ -0,0 +1,128 @@
+    DEFAULT_SUMMARY_FILE = "md_summary.yml"
+
+    @classmethod
+    def define(cls, spec: CalcJobProcessSpec) -> None:
+        """
+        Define the process specification, its inputs, outputs and exit codes.
+
+        Parameters
+        ----------
+        spec : `aiida.engine.CalcJobProcessSpec`
+            The calculation job process spec to define.
+        """
+        super().define(spec)
+
+        # Additional inputs for molecular dynamics
+        spec.input(
+            "ensemble",
+            valid_type=Str,
+            required=False,
+            help="Name for thermodynamic ensemble",
+        )
+
+        spec.input(
+            "md_kwargs",
+            valid_type=Dict,
+            required=False,
+            default=lambda: Dict(
+                {
+                    "traj-file": cls.DEFAULT_TRAJ_FILE,
+                    "stats-file": cls.DEFAULT_STATS_FILE,
+                    "summary": cls.DEFAULT_SUMMARY_FILE,
+                }
+            ),
+            help="Keywords for molecular dynamics",
+        )
+
+        spec.inputs["metadata"]["options"]["parser_name"].default = "janus.md_parser"
+
+        spec.output(
+            "results_dict",
+            valid_type=Dict,
+            help="The `results_dict` output node of the successful calculation.",
+        )
+        spec.output("summary", valid_type=SinglefileData)
+        spec.output("stats_file", valid_type=SinglefileData)
+        spec.output("traj_file", valid_type=SinglefileData)
+        spec.output("traj_output", valid_type=TrajectoryData)
+        spec.output("final_structure", valid_type=StructureData)
+
+        spec.default_output_node = "results_dict"
+
+    def prepare_for_submission(
+        self, folder: aiida.common.folders.Folder
+    ) -> datastructures.CalcInfo:
+        """
+        Create the input files for the `Calcjob`.
+
+        Parameters
+        ----------
+        folder : aiida.common.folders.Folder
+            An `aiida.common.folders.Folder` to temporarily write files on disk.
+
+        Returns
+        -------
+        aiida.common.datastructures.CalcInfo
+            An instance of `aiida.common.datastructures.CalcInfo`.
+        """
+
+        # Call the parent class method to prepare common inputs
+        calcinfo = super().prepare_for_submission(folder)
+        codeinfo = calcinfo.codes_info[0]
+
+        md_dictionary = self.inputs.md_kwargs.get_dict()
+
+        md_dictionary.setdefault("traj-file", str(self.DEFAULT_TRAJ_FILE))
+        md_dictionary.setdefault("stats-file", str(self.DEFAULT_STATS_FILE))
+        md_dictionary.setdefault("summary", str(self.DEFAULT_SUMMARY_FILE))
+
+        if "ensemble" in self.inputs:
+            ensemble = self.inputs.ensemble.value.lower()
+        elif "ensemble" in self.inputs.config.as_dictionary.items():
+            ensemble = self.inputs.config.as_dictionary["ensemble"]
+        else:
+            print("ERROR ENSEMBLE NOT PROVIDED")
```

Comment:
this one was for debugging, I forgot to remove it. I think rather than this I will put some input validation to ensure that somewhere the ensemble is given, does it sound better?
but if janus has a default for the ensemble I'll just make it use the default

Reply:
There's no default at the moment, although it's something we have briefly considered. For now, yes catching it around here would be nice